clients/elrond/client.go

```diff
@@ -335,32 +336,40 @@ func (c *client) PerformAction(ctx context.Context, actionID uint64, batch *clie
 	return hash, err
 }
 
-//CheckClientAvailability will check the client availability and will set the metric accordingly
+// CheckClientAvailability will check the client availability and will set the metric accordingly
 func (c *client) CheckClientAvailability(ctx context.Context) error {
 	c.mut.RLock()
 	defer c.mut.RUnlock()
+
 	currentNonce, err := c.GetCurrentNonce(ctx)
 	if err != nil {
-		c.statusHandler.SetStringMetric(bridgeCore.MetricElrondClientStatus, ethElrond.Unavailable.String())
-		c.statusHandler.SetStringMetric(bridgeCore.MetricLastElrondClientError, err.Error())
+		c.setStatusForAvailabilityCheck(ethElrond.Unavailable, err.Error())
+
 		return err
 	}
-	c.statusHandler.SetStringMetric(bridgeCore.MetricLastElrondClientError, "")
 
 	if currentNonce != c.lastNonce {
 		c.retriesAvailabilityCheck = 0
-		c.statusHandler.SetStringMetric(bridgeCore.MetricElrondClientStatus, ethElrond.Available.String())
+		c.setStatusForAvailabilityCheck(ethElrond.Available, "")
```

Comment:
i would still empty MetricLastElrondClientError independently of the nonce. 
If we had previously one error and next time we query the nonce, it gives it without error, but it is the same as the store one. We would still have the error (that is not valid anymore) stored in the metric

Reply:
maybe also add a test for this case

Reply:
done & done